ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OGridSearch.scala

```diff
@@ -176,11 +176,15 @@ class H2OGridSearch(override val uid: String)
           H2OBinaryModel.read("file://" + downloadedModel.getAbsolutePath, Some(modelId))
       }
     } else {
-      sortedGridModels.map { case (modelId, _) => H2OModel(modelId).tryDelete() }
+      sortedGridModels.foreach { case (modelId, _) => H2OModel(modelId).tryDelete() }
     }
     algo.deleteRegisteredH2OFrames()
     deleteRegisteredH2OFrames()
-    gridModels.head
+    val result: H2OMOJOModel = gridModels.head
+    if (H2OContext.get().forall(_.getConf.isModelPrintAfterTrainingEnabled)) {
```

Comment:
Proposal: The method [`getGridModels()`](https://github.com/h2oai/sparkling-water/pull/2715/files#diff-2813143693098236fae6163182f97f766f262fa5d827764e1ca73afb7565b8d5R163) accesses h2oContext. I think you can get the h2oContext in the fit method and use it on two places (here, pass h2oContext to `getGridModels()`